File: pinyougou_parent/pinyougou_shop_web/src/main/webapp/js/service/goodsService.js
app.service('goodsService', function ($http) {
    //旧的查询方法(弃置)								   
    this.findPage = function (pageNum, pageSize) {
        return $http.get('../goods/findPage.do?pageNum=' + pageNum + '&pageSize=' + pageSize)
    };

    //查询方法,读取列表数据绑定到表单中
    this.search = function (pageNum, pageSize, searchEntity) {
        return $http.post('../goods/findPage.do?pageNum=' + pageNum + "&pageSize=" + pageSize, searchEntity)
    };

    //删除方法
    this.removeGoodsList = function (idList) {
        return $http.get('../goods/removeGoods.do?idList=' + idList)
    };

    //根据id查询
    this.findOne = function (id) {
        return $http.get('../goods/findOne.do?id=' + id)
    };

    //修改品牌
    this.updateGoods = function (goods) {
        return $http.post('../goods/updateGoods.do', goods)
    };

    //新增品牌
    this.save = function (goodsGroup) {
        return $http.post('../goods/saveGoods.do', goodsGroup)
    };   
});

File: pinyougou_parent/pinyougou_sellergoods_interface/src/main/java/com/pinyougou/sellergoods/service/GoodsService.java
package com.pinyougou.sellergoods.service;
import java.util.List;

import com.pinyougou.group.GoodsGroup;
import com.pinyougou.pojo.TbGoods;
import entity.PageResult;
import entity.Result;
/**
 * goods服务层接口
 * @author jieweili
 *
 */
public interface GoodsService {

    /**
     * 返回全部列表(已弃置)
     * @return
     */
    List<TbGoods> findAll();
 
    /**
     * 返回分页对象(过时)
     * @param pageNum
     * @param pageSize
     * @return
     */
	 
    PageResult findPage(Integer pageNum, Integer pageSize);

    /**
     * 添加数据 组合类
     * @param goodsGroup
     * @return
     */
    public Result saveGoods(GoodsGroup goodsGroup);

    /**
     * 根据id查询一个对象
     * @param id
     * @return
     */
    TbGoods getGoodsById(Long id);

    /**
     * 修改对象
     * @param goods
     * @return
     */
    Result updateGoods(TbGoods goods);

    /**
     * 批量删除
     * @param idList
     * @return
     */
    Result removeGoods(Long[] idList);

    /**
     * 根据条件分页查询
     * @param goods
     * @param pageNum
     * @param pageSize
     * @return
     */
    PageResult listGoodsByExample(TbGoods goods, Integer pageNum, Integer pageSize);


 
}
